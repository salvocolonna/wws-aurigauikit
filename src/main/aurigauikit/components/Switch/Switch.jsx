import React from "react"
import "./switch.less"

export default ({ checked = false, onChange }) => (
  <label className="switch">
    <input type="checkbox" value={checked} onChange={onChange} />
    <span className="slider round" />
  </label>
)
