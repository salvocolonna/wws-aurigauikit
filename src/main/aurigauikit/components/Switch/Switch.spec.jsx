import React from "react"
import renderer from "react-test-renderer"
import { shallow } from "enzyme"
import Switch from "./Switch"

describe("Switch", () => {
  it("displays as expected", () => {
    const component = renderer.create(<Switch checked />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("calls onChange callback", () => {
    const onChange = jest.fn()
    const component = shallow(<Switch onChange={onChange} />)
    component.find("input").simulate("change")
    expect(onChange.mock.calls).toHaveLength(1)
  })
})
