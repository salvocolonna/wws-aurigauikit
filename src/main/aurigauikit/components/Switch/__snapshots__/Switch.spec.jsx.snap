// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch displays as expected 1`] = `
<label
  className="switch"
>
  <input
    onChange={undefined}
    type="checkbox"
    value={true}
  />
  <span
    className="slider round"
  />
</label>
`;
